torch/_dynamo/guards.py

```diff
@@ -3357,6 +3372,55 @@ def get_guard_fail_reason(
             "Failure in guard_fail_fn callback - raising here will cause a NULL Error on guard eval",
         )
 
+    return reason_str, code_reasons, fail_reasons
+
+
+def _extract_recompiled_dynamic_sources(
+    fail_reasons: list[str], frame_locals: dict[str, object], guard_manager: GuardFn
+) -> tuple[list[str], bool]:
+    """
+    Goes through a list of guard failure reasons, and extracts source names for tensors we've dynamically recompiled.
```

Comment:
Hmmmm so one downside of this opt-in is by default users won't get this for their first mast run?